### ISSUE TITLE
Rule proposal: default-break

### ISSUE THREAD

**Issue by adroitwhiz** [eslint/eslint#12991]
**Please describe what the rule should do:**

Require or forbid (depending on configuration) the `break` statement in the `default` case of a `switch` statement.

**What category of rule is this? (place an "X" next to just one item)**

- [ ] Warns about a potential error (problem)
- [ ] Suggests an alternate way of doing something (suggestion)
- [x] Enforces code style (layout)
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

`"default-break": "always"`:
```js
switch (val) {
  case 1:
    doSomething();
    break;
  case 2:
    doADifferentThing();
    break;
  default:
    doTheDefaultThing(); // error: missing `break` statement
}
```

`"default-break": "never"`:
```js
switch (val) {
  case 1:
    doSomething();
    break;
  case 2:
    doADifferentThing();
    break;
  default:
    doTheDefaultThing();
    break; // error: unnecessary `break` statement
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**
This rule is very small in scope and covers a relatively common construct in the core language. Most ESLint plugins I've seen implement rules for third-party language extensions (e.g. JSX) or are much larger in scope (e.g. forbidding all mutability). If ESLint users had to install a plugin for this one rule, it would probably not be widely adopted.

[This is essentially a matter of style](https://softwareengineering.stackexchange.com/questions/201777/break-on-default-case-in-switch), with most programmers leaning towards including the `break` statement. It would be useful to be able to enforce it with a linter.

**Are you willing to submit a pull request to implement this rule?**
If I can figure out the codebase.

**Comment by mdjermanovic** [comment]
Hi @adroitwhiz, thanks for this proposal!

We'll have `switch-final-break` rule (accepted PR #12094) to require/disallow `break` in the last clause, which can be `case` or `default`.
